### PR TITLE
Move Odroid_C2 to Amlogic Project

### PR DESCRIPTION
**Pull request by Raybuntu**
This is based on @adamg88 PR https://github.com/LibreELEC/LibreELEC.tv/pull/2391
Move C2 project to be a device inside the Amlogic Project.
I've also reworked the boot.ini. @adamg88 already updates boot.ini in update.sh but all user settings would be overriden which is bad. However we should be able to change some setting in boot.ini in the future for instance if there is a new Feature or Bug to fix. For instance I've added a routine to enable/disable RTC Shield in boot.ini. That way we can get rid of those spam messages in dmesg (see https://forum.libreelec.tv/thread/8789-pcf8563-get-datetime-read-error/).
Also I'm introducing a new /flash/user.ini. Users can for instance enable RTC Shield in there and it won't be overriden on the next update. I've copied most of the comments over to user.ini for settings that can be changed by Users. I've cleaned up all the other non working stuff in there as well.

The linux.conf change is a preperation for Crazycat. See https://github.com/LibreELEC/LibreELEC.tv/pull/2405
I'd prefer this PR to go in before https://github.com/LibreELEC/LibreELEC.tv/pull/2405.

I've tested it on my C2 (clean build, fresh flash + tar update). The only thing I haven't tested is CEC which I will do this Weekend. I also have to test a LE8 C2 tar update. 
The first update from previous C2 builds will override the boot.ini. It means if the user had any settings in there it will be lost. Maybe we should create a conditional backup of that old boot.ini.

### CODE REVIEW

File: projects/Amlogic/devices/Odroid_C2/bootloader/user.ini [hunk 1]
```diff
@@ -0,0 +1,112 @@
+#------------------------------------------------------------------------------------------------------
```
Comment (adamg88) [inline projects/Amlogic/devices/Odroid_C2/bootloader/user.ini]: I only have 1 suggestion, perhaps this should be called boot.user.ini as user.ini is somewhat vague.

and then add a header to boot.ini saying do not modify, changes may be lost.

File: projects/Amlogic/filesystem/usr/share/bootloader/update.sh [hunk 2]
```diff
@@ -98,6 +98,14 @@ if [ -f $SYSTEM_ROOT/usr/share/bootloader/boot.ini ]; then
   cp -p $SYSTEM_ROOT/usr/share/bootloader/boot.ini $BOOT_ROOT/boot.ini
 fi
 
+if [ -f $SYSTEM_ROOT/usr/share/bootloader/user.ini ]; then
```
Comment (kszaq) [inline projects/Amlogic/filesystem/usr/share/bootloader/update.sh]: Please merge this with the case above it.

File: projects/Amlogic/initramfs/platform_init [hunk 3]
```diff
@@ -110,3 +110,9 @@ echo 8 > /proc/irq/62/smp_affinity   # dwc_otg, dwc_otg_hcd:usb2, dwc_otg_pcd
 echo 4 > /sys/module/amvdec_h264/parameters/dec_control
 
 echo 1 > /sys/devices/system/cpu/cpufreq/ondemand/io_is_busy
+
+# Turn off Blue Flashing LED on Odroid C2
```
Comment (kszaq) [inline projects/Amlogic/initramfs/platform_init]: I think we don't need this, we can simply patch the device tree.

File: projects/Amlogic/linux/linux.aarch64.conf [hunk 4]
```diff
@@ -404,7 +404,7 @@ CONFIG_PM_SLEEP=y
 CONFIG_PM_SLEEP_SMP=y
 # CONFIG_PM_AUTOSLEEP is not set
 # CONFIG_PM_WAKELOCKS is not set
-# CONFIG_PM_RUNTIME is not set
+CONFIG_PM_RUNTIME=y
```
Comment (kszaq) [inline projects/Amlogic/linux/linux.aarch64.conf]: Why is this change needed? IIRC this was turned off because of some issues with USB devices.

Comment (Raybuntu) [reply projects/Amlogic/linux/linux.aarch64.conf]: This is preparing for crazycat

File: projects/Amlogic/devices/Odroid_C2/bootloader/user.ini (hunk 1 above)
Comment (Raybuntu) [reply projects/Amlogic/devices/Odroid_C2/bootloader/user.ini]: The header is fine. I'm not sure about the user ini filename yet.

File: projects/Amlogic/initramfs/platform_init (hunk 3 above)
Comment (Raybuntu) [reply projects/Amlogic/initramfs/platform_init]: Didn't know we can. Can you give me a hint in dtb please.

File: projects/Amlogic/filesystem/usr/share/bootloader/update.sh (hunk 2 above)
Comment (Raybuntu) [reply projects/Amlogic/filesystem/usr/share/bootloader/update.sh]: Can do.

File: projects/Amlogic/devices/Odroid_C2/bootloader/user.ini (hunk 1 above)
Comment (chewitt) [reply projects/Amlogic/devices/Odroid_C2/bootloader/user.ini]: pi has config.txt so perhaps config.ini ?

**Comment by Raybuntu** [comment]
I have rebased to fix all suggestions.

File: projects/Amlogic/devices/S905/options [hunk 5]
```diff
@@ -2,4 +2,11 @@
   # for a list of additinoal drivers see packages/linux-drivers
   # Space separated list is supported,
   # e.g. ADDITIONAL_DRIVERS="DRIVER1 DRIVER2"
-    ADDITIONAL_DRIVERS="$ADDITIONAL_DRIVERS fd628-aml"
+    ADDITIONAL_DRIVERS="$ADDITIONAL_DRIVERS ap6xxx-aml mt7601u-aml mt7603u-aml \
```
Comment (afl1) [inline projects/Amlogic/devices/S905/options]: You have to fix also S912 device.

**Comment by Raybuntu** [comment]
@afl1 Sorry for not fixing this sooner. I've been not feeling well the last couple of days.
@kszaq I've also made all the required changes you told me on slack. Let me know if there is more.
